akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbCompaction.scala

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.persistence.journal.leveldb
+
+import akka.actor.Actor
+
+import scala.concurrent.Future
+
+private[persistence] object LeveldbCompaction {
+
+  case class TryCompactLeveldb(persistenceId: String, toSeqNr: Long)
+}
+
+/**
+ * INTERNAL API.
+ *
+ * Exposure of LevelDB compaction capability to reduce journal size upon message deletions.
+ */
+private[persistence] trait LeveldbCompaction extends Actor with CompactionSegmentManagement { this: LeveldbStore ⇒
+
+  import Key._
+  import LeveldbCompaction._
+
+  def receiveCompactionInternal: Receive = {
+    case TryCompactLeveldb(persistenceId, toSeqNr) ⇒
+      tryCompactOnDelete(persistenceId, toSeqNr)
+  }
+
+  private def tryCompactOnDelete(persistenceId: String, toSeqNr: Long): Unit = {
+    if (mustCompact(persistenceId, toSeqNr)) {
+      val limit = compactionLimit(persistenceId, toSeqNr)
+      //      System.err.println(String.format(
+      //        "Starting compaction for id [%s] up to seqNo [%s] with limit [%s]",
+      //        persistenceId, toSeqNr.toString, limit.toString))
```

Comment:
I left these comments there on purpose actually for us to discuss this. I actually wanted to add some logging here, but I noticed that ActorLogging was not already mixed into LeveldbJournal, so I was reluctant to do so. Do you confirm that it is OK for me to mix it in?